src/main/webapp/profile.js

```diff
@@ -125,89 +142,188 @@ function editProfile(user, role, document) {
     document.getElementById('profile-container').style.display = 'none';
     document.getElementById('edit-profile-form').style.display = 'block';
 
-    // Set default value of bio to what user previously had
-    // This will prevent this data from being lost if the user decides not change this field
-    document.getElementById('bio').value = user.bio;
+    if (role == "both") {
+        // Set default value of bio to what user previously had
+        // This will prevent this data from being lost if the user decides not change this field
+        document.getElementById('bio').value = user.student.bio;
 
-    if (role == 'student') {
-        document.getElementById('student-topics').style.display = 'block';
+        editProfileTutor(user, true, document);
+        editProfileStudent(user, true, document);
+    } else if (role == 'student') {
+        document.getElementById('bio').value = user.bio;
+
+        editProfileStudent(user, false, document);
+    } else if (role == 'tutor') {
+        document.getElementById('bio').value = user.bio;
+
+        editProfileTutor(user, false, document);
+    }
+}
+
+function editProfileTutor(user, both, document) {
+    if (both == true) {
```

Comment:
```suggestion 
    var tutor;
    if (both == true) {
        tutor = user.tutor;
    } else {
        tutor = user;
    }
    
    document.getElementById('tutor-topics').style.display = 'block';
        // Automatically check boxes of topics user has already selected
        if (tutor.skills.indexOf('math') > -1) {
            document.getElementById('math-tutor').checked = true;
        }
        if (.tutor.skills.indexOf('physics') > -1) {
            document.getElementById('physics-tutor').checked = true;
        }
        if (tutor.skills.indexOf('chemistry') > -1) {
            document.getElementById('chemistry-tutor').checked = true;
        }
        if (tutor.skills.indexOf('biology') > -1) {
            document.getElementById('biology-tutor').checked = true;
        }
        if (tutor.skills.indexOf('computer science') > -1) {
            document.getElementById('computer-science-tutor').checked = true;
        }
        if (tutor.skills.indexOf('social studies') > -1) {
            document.getElementById('social-studies-tutor').checked = true;
        }
        if (tutor.skills.indexOf('english') > -1) {
            document.getElementById('english-tutor').checked = true;
        }
        if (tutor.skills.indexOf('spanish') > -1) {
            document.getElementById('spanish-tutor').checked = true;
        }
        if (tutor.skills.indexOf('french') > -1) {
            document.getElementById('french-tutor').checked = true;
        }
        if (tutor.skills.indexOf('chinese') > -1) {
            document.getElementById('chinese-tutor').checked = true;
        }
        var otherTopicsIndex = tutor.skills.indexOf(' ');
        var otherTopics = [];
        // Check if any other entries exist
        if (otherTopicsIndex > -1 && tutor.skills.length > otherTopicsIndex + 1) {
            otherTopics = tutor.skills.slice(otherTopicsIndex + 1);
        }
        document.getElementById('other-subject-tutor').value = otherTopics.join(', ');
    
    
```

Reply:
Sorry, the indentation got messed up! 
Instead of repeating the if statements for user and user.tutor, we can just store whichever we need in a variable called `tutor` and only have to write the if statements once.